tp1/tp1.go

```diff
@@ -20,21 +25,120 @@ type Carrito struct {
 	Precio int
 }
 
+type producto struct {
+	id, precio int
+}
+
+func (p producto) ID() int {
+	return p.id
+}
+
+func (p producto) Precio() int {
+	return p.precio
+}
+
 // CalcularPrecios recibe un arreglo de los IDs de productos y calcula,
 // para cada super mercado, cuanto saldria comprar esos productos ahi.
 // Retorna un slice de carritos, donde se tiene uno para cada super mercado.
 func (p Productos) CalcularPrecios(ids ...int) []Carrito {
-	return nil
+
+	// No hay ningun producto
+	if len(ids) == 0 {
+		return nil
+	}
+
+	var carritos []Carrito
+	for _, s := range Supermercados {
+		carritos = append(carritos, Carrito{
+			Tienda: s,
+		})
+	}
+
+	for _, producto := range p {
+		for _, id := range ids {
+			if pID := strToInt(producto[1]); pID == id {
+				/* Este for no me gusta pero si utilizo un range
+				   se me va a generar una copia de carritos
+				   y no estaría acumulando los precios.
+				   Entiendo que si carritos fuese del tipo *[]Carrito
+				   podría pasar el valor por referencia
+				*/
+				for c := 0; c < len(carritos); c++ {
+					if producto[0] == carritos[c].Tienda {
+						p, _ := strconv.Atoi(producto[2])
+						carritos[c].Precio += p
+					}
+				}
+			}
+		}
+	}
+
+	return carritos
 }
 
 // Promedio recibe el id de un producto y retorna el precio promedio
 // de ese producto usando los precios de todos los supermercados.
 func (p Productos) Promedio(idProducto int) float64 {
-	return 0
+
+	var totalPrecio, cantProducto float64
+
+	for _, producto := range p {
+		if pID := strToInt(producto[1]); pID == idProducto {
+			pPrecio := strToFloat(producto[2], 64)
+			totalPrecio += pPrecio
+			cantProducto++
+		}
+	}
+
+	// El producto ingresado no existe
+	if totalPrecio == 0 {
+		return totalPrecio
+	}
+
+	return totalPrecio / cantProducto
 }
 
 // BuscarMasBarato recibe un id de producto a buscar y te retorna
 // el producto mas barato que haya encontrado.
 func (p Productos) BuscarMasBarato(idProducto int) (Producto, bool) {
-	return nil, false
+
+	prodBarato := producto{idProducto, 0}
+	esPrimeraIteracion := true
+
+	for _, prod := range p {
+		if pID := strToInt(prod[1]); pID == idProducto {
```

Comment:
Lo mismo en este caso. Si utilizamos la condicion de corte vamos a tener nuestro codigo significante con menor indentacion. Mejora lo que en go le dicen "line of sight". https://youtu.be/yeetIgNeIkc?t=257

Reply:
Implementado. Muchas gracias por tomarte el tiempo en dar feedback e incluso haber citado una conf. Voy a mirarlo. Gracias nuevamente.